### PR TITLE
dash-test-workflow-p4-saithrift

### PR DESCRIPTION
**Pull request by mmiele**
Rebuilt file and added diagram.
It replaces the PR: https://github.com/Azure/DASH/pull/25

### CODE REVIEW

**Review by chrispsommers — CHANGES_REQUESTED** [review]
1. Check spelling for "saithrift" - it says "saihrift"
2. I see a stray + sign near center left
3. "libsai" blue boxes - the text seems blurry, is it a bit-map or native SVG filled box w/ text?

File: test/docs/dash-test-workflow-p4-saithrift.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+A libsai library will be developed which maps SAI conceptual APIs into the [sirius_pipeline.p4](../sirius_pipeline.p4) TDI implementation, all done in-process. This is very efficient because it avoids serialization over P4Runtime.
+
+The P4-DPDK dataplane also has a built-in P4Runtime server which can be used develop and test the P4 Model. This is not part of the official DASH test workflow.
+
+The P4-DPDK dataplane is bound to host veth ports at startup.  These are "wired" to the software traffic generator using Linux bridging or similar.
+
+## Summarizing it all
+* Standard DASH test cases and test runners apply conformance and performance tests against software-based DASH dataplane simulators.
+* Software dataplane have lower scale and performance capabilities compared to production hardware implementations. Therefore, test must be marked for suitability on each target to respect performance and scale limitations of each software implementation. The test runners will run only the appropriate subset for a given target.
```
Comment (chrispsommers) [inline test/docs/dash-test-workflow-p4-saithrift.md:74]: s/test must be marked/tests must be marked/  <-- for future reference, this is a SW dev's way to show a string substitution i.e. correct a typo

Comment (mmiele) [reply test/docs/dash-test-workflow-p4-saithrift.md:74]: @chrispsommers Could you clarify the remark you made? It is not clear what I need to do. Thanks

**Comment by mmiele** [comment]
> 
Check spelling for "saithrift" - it says "saihrift"
I see a stray + sign near center left
"libsai" blue boxes - the text seems blurry, is it a bit-map or native SVG filled box w/ text?

Fixed the diagram.

**Comment by chrispsommers** [comment]
HI Michael, My comment means "please fix my typo by changing test to tests in the phrase. Thanks.

File: Assets/cplus-saithrift-libsai.svg [hunk 2]
```diff
@@ -0,0 +1 @@
...EzY9G258Ci0crbmF3N0lRK08zT3EMamkBAYXgoZiQQMxAEGBgCtdBptxd0loAAAAABJRU5ErkJggg==" preserveAspectRatio="none" id="img8"></image><clipPath id="clip9"><rect x="0" y="0" width="381000" height="200186"/></clipPath></defs><g clip-path="url(#clip0)" transform="translate(-685 -290)"><g clip-path="url(#clip1)" transform="matrix(0.000104987 0 0 0.000104987 692 290)"><g clip-path="url(#clip3)" transform="matrix(1 0 0 1.46202 -0.5 -0.25)"><use width="100%" height="100%" xlink:href="#img2" transform="scale(8903.8 8903.8)"></use></g></g><text fill="#FFFFFF" font-family="Calibri,Calibri_MSFontService,sans-serif" font-weight="700" font-size="12" transform="translate(695.576 326)">saihrift</text><g clip-path="url(#clip4)" transform="matrix(0.000104987 0 0 0.000104987 705 293)"><g clip-path="url(#clip6)" transform="matrix(1 0 0 1.1363 0 -0.25)"><use width="100%" height="100%" xlink:href="#img5" transform="scale(4497.92 4497.92)"></use></g></g><g clip-path="url(#clip7)" transform="matrix(0.000104987 0 0 0.000104987 735 299)"><g clip-path="url(#clip9)" transform="matrix(1 0 0 1.18952 0 -0.25)"><use width="100%" height="100%" xlink:href="#img8" transform="scale(6457.63 6457.63)"></use></g></g></g></svg>
```
[the remark this answers is not in the document]
Comment (mmiele) [reply Assets/cplus-saithrift-libsai.svg:1]: Fixed.

File: Assets/icons/snappi-icon.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...7O0tbW1tre6u7y8vcTGx8nKy8zMztDQ0NXW19fY2Nna4eLo7vH09Pb4+vz8/P39/u+M3bcAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAHVSURBVChTY0AGOqvMGRiMVxtCucQDkxu/rloZXvp02xoqgAfwxIJIEXufQMfE1LhDP798P7jrw9uPO8GSeIHoHBDJJiErL6WhoZ77+9vjwsyH7z/nMzDo1zY0lAs7VzU0NEWxRDYCKU+G7IaGhno9p8rGhsZInhQGViFpJTlVVWVJoBEZd+sYGMru5TEwCFzetm7dvv2Trq9Zt+acbfM6ILiodWXj2nVbT068tnbd2pshZxjEzTxigtOScrZrMzCwr+AH6rdjYmBQuaPAwMBnOakByO+IADmMYYvRHg6g2Plp/UDOzJpTYEEGBpnZz09oKm74c1gNwufYe+TAgb3pkxqB7E6/9gNAzi09kE7hs2CdM6pPMfhHxwPBpjdvXx1b/+PL6x1cYJ0MCi6urlkPpoB1ei13d3V12W2wmxVo59lpfUAxkE5dUxsbG4u2R++ezup58fXZZBawRuH54QEBpfentQDZXb4LUwICAo4aHA8KCAi7MHElkLO55DRYHQMDZ8WTubyCrS+ni0H4jEVLli5d4F0QCmQnOLgtWrp08TyVk4uXLl2S3H1u8dJlExR6IQoZGLiLgU5hzoDysAOQaxkYpnWDOSSBvcAQYmCYCgohBgYGAGAYrjWPlly8AAAAAElFTkSuQmCC" preserveAspectRatio="none" id="img2"></image><clipPath id="clip3"><rect x="0" y="0" width="542925" height="109805"/></clipPath></defs><g clip-path="url(#clip0)" transform="translate(-675 -392)"><g clip-path="url(#clip1)" transform="matrix(0.000104987 0 0 0.000104987 676 392)"><g clip-path="url(#clip3)" transform="scale(1 1.04094)"><use width="100%" height="100%" xlink:href="#img2" transform="scale(6100.28 6100.28)"></use></g></g></g></svg>
```
[the remark this answers is not in the document]
Comment (mmiele) [reply Assets/icons/snappi-icon.svg:1]: Got original logo.

**Review by reshmaintel — COMMENTED** [review]
Reviewed, changes have been incorporated.

**Pull request merged**